.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,44 @@
+name: Integration Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  GRADLE_OPTS: -Dorg.gradle.daemon=false -Xmx6g -Xms6g
+
+jobs:
+  it-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    # Install Java 8 for cross-compilation support. Setting it up before
+    # Java 11 means it comes later in $PATH (because of how setup-java works)
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Cache on push
+      if: github.event_name == 'push'
+      uses: actions/cache@v2
+      with:
+        path: ~/.gradle
+        key: ${{ runner.os }}-cd-it-${{ github.sha }}
+        # Restore build outputs from the previous commit (if successful), if current commit hasn't run successfully yet
+        restore-keys: |
+          ${{ runner.os }}-cd-it-${{ github.event.before }}
+    - name: Cache on pull_request
```

Comment:
How are we specifying that this check is non-blocking to PRs? If that has to be done later/separately, that's cool, just wondering if I missed something.